### PR TITLE
Remove helm patches to vendor

### PR DESCRIPTION
**Pull request by pacevedom**
Remove patches to helm by removing unneeded code and/or copying logic.

### CODE REVIEW

**Comment by pacevedom** [comment]
cc @qbarrand

File: pkg/helmer/helmer.go [hunk 1]
```diff
@@ -163,6 +164,43 @@ func (h *helmer) logWrap(format string, v ...interface{}) {
 	h.log.Info("Helm", "internal", msg)
 }
 
+func (h *helmer) failRelease(rel *release.Release, err error) error {
```
Comment (qbarrand) [inline pkg/helmer/helmer.go:167]: The [Helm code](https://github.com/helm/helm/blob/main/pkg/action/install.go#L435) also uninstalls the release. Don't we want to replicate that behaviour in SRO?

**Review by qbarrand — COMMENTED** [review]
This is really great work. Let's hope this is temporary until we can rework the chart installation flow. Thanks!

File: pkg/helmer/helmer.go (hunk 1 above)
Comment (pacevedom) [reply pkg/helmer/helmer.go:167]: This is only done when the release is configured to be atomic, which is either install everything, or nothing. Not our case as per https://github.com/openshift-psap/special-resource-operator/blob/master/pkg/helmer/helmer.go#L202-L211

**Pull request merged**